jj/mock/_history/_history_representation.py

```diff
@@ -0,0 +1,71 @@
+import ast
+import json
+import os
+import shutil
+from abc import ABC, abstractmethod
+from pprint import pformat
+from typing import List, Optional
+
+from jj.mock._history._history_item import HistoryItem
+
+__all__ = ("default_history_formatter", "HistoryFormatter", "PrettyHistoryFormatter",
+           "SimpleHistoryFormatter",)
+
+
+REMOTE_MOCK_PPRINT_LIMIT = os.environ.get("JJ_REMOTE_MOCK_PPRINT_LIMIT", "1000000")
+REMOTE_MOCK_PPRINT_WIDTH = os.environ.get("JJ_REMOTE_MOCK_PPRINT_WIDTH")
+
+
+class HistoryFormatter(ABC):
+    @abstractmethod
+    def format_history(self, history: Optional[List[HistoryItem]]) -> str:
+        pass
+
+
+class SimpleHistoryFormatter(HistoryFormatter):
+    def __init__(self) -> None:
+        self.history_output_width = None
+
+    def format_history(self, history: Optional[List[HistoryItem]]) -> str:
+        parsed_history = [{"req": x["request"].to_dict(),
+                           "res": x["response"].to_dict()} for x in history] if history else []
+        history_as_strings = [json.dumps(x, default=str) for x in parsed_history]
+        formatted_history = '\n'.join(history_as_strings)
+        return formatted_history
+
+
+class PrettyHistoryFormatter(HistoryFormatter):
+    def __init__(self, history_output_limit: Optional[int] = None,
+                 history_output_width: Optional[int] = None) -> None:
+        if not history_output_limit:
+            history_output_limit = int(REMOTE_MOCK_PPRINT_LIMIT) if REMOTE_MOCK_PPRINT_LIMIT \
+                else 1000000
+        if not history_output_width:
+            history_output_width = int(REMOTE_MOCK_PPRINT_WIDTH) if REMOTE_MOCK_PPRINT_WIDTH \
+                else shutil.get_terminal_size((80, 20))[0]
+        self._history_output_limit = history_output_limit
+        self.history_output_width = history_output_width
+
+    def __cut_str__(self, string: str, length: int, separator: str = "..") -> str:
+        assert length > len(separator)
+        if len(string) <= length:
+            return string
+        length -= len(separator)
+        return string[:length // 2] + separator + string[-length // 2:]
+
+    def format_history(self, history: Optional[List[HistoryItem]]) -> str:
+        parsed_history = [{"req": x["request"].to_dict(),
+                           "res": x["response"].to_dict()} for x in history] if history else []
+        history_as_strings = [self.__cut_str__(string=json.dumps(x, default=str),
+                                               length=self._history_output_limit)
+                              for x in parsed_history]
+        history_objects = [ast.literal_eval(history_string)
```

Comment:
Your solution appears to be overcomplicated. A simpler approach could involve using just the `pformat` function:

```python
parsed_history = [...]
return "\n".join([
    self.__cut_str__(
        string=pformat(x, width=self.history_output_width),
        length=self._history_output_limit
    )
    for x in parsed_history
])
```

Reply:
Fixed